### ISSUE TITLE
AWS API GW should support AWS_PROXY integration type.

### ISSUE THREAD

**Issue by BerndWessels** [hashicorp/terraform#9249]
Hi there,

we desperately need `AWS_PROXY` as type for our `API Gateway integration` to the `Lambda function`.

AWS actually supports that, but terraform seems not to be updated with the new types.

### Terraform Version
v0.7.4

### Affected Resource(s)
- aws_api_gateway_integration

### Terraform Configuration Files
```hcl
resource "aws_api_gateway_integration" "SeoWebsite" {
  rest_api_id = "${aws_api_gateway_rest_api.SeoWebsite.id}"
  resource_id = "${aws_api_gateway_resource.SeoWebsite.id}"
  http_method = "${aws_api_gateway_method.SeoWebsite.http_method}"
  type = "AWS_PROXY"
  uri = "arn:aws:apigateway:${var.aws_region}:lambda:path/2015-03-31/functions/${aws_lambda_function.SeoWebsite.arn}/invocations"
  integration_http_method = "${aws_api_gateway_method.SeoWebsite.http_method}"
}
```

### Expected Behavior
Allow `AWS_PROXY` value for `type`.

### Actual Behavior
`Errors: aws_api_gateway_integration.SeoWebsite: "type" must be one of 'AWS', 'MOCK', 'HTTP'`

### References
- https://aws.amazon.com/blogs/aws/api-gateway-update-new-features-simplify-api-development/

**Comment by kwilczynski** [comment]
@BerndWessels hi there! I have a very good news for you!

The work on adding this has already been done, and is due to release together with next point-release, as per: https://github.com/hashicorp/terraform/pull/9213

Keep an eye on an upcoming release.

**Comment by stack72** [comment]
Closed via #9213 :)
